### DISCUSSION TITLE
Feature Request: shutdown by file name

### DISCUSSION THREAD

**Discussion by Jorges1000** [duckdb/duckdb#11728] [Ideas]
I have scripts calling duck db, e.g.

```
try(duckdb::dbDisconnect(ddcon,shutdown=TRUE)
ddcon <- duckdb::dbConnect(duckdb::duckdb('dd.ddb'))
```
The reason it is preceded by a shutdown is because if the second line has already been called, it would give the "cannot access the file because it is being used by another process". But this only works if ddcon exists in memory. As far as I know, there is not way to shutdown the connection if it was generated by another script or user and ddcon is not available or refers to a different connection, other than restarting R.

I understand it is not good practice to shutdown an existing connection and mess up someone else's work, but in our use case each user should get their own DuckDb file. 

Would be nice to have a function such as `duckdb_file_shutdown(dbdir='dd.ddb')` that would shutdown any connections to dbdir.

**Comment by soerenwolfers** [comment]
That's likely better handled by implementing the required inter process communication in your own processes, similar to how if you have two python scripts writing and reading from the same parquet file, you wouldn't expect the parquet format to handle your priority preferences for you.

**Reply by Jorges1000** [reply]
Shutting down R is the only solution if there is a .ddb.wal file from a previous connection that was not shutdown and the driver object is not available. There must be a better way.
